app/src/shared/utils/web3.js

```diff
@@ -28,3 +28,17 @@ export const hasTransactionCompleted = (txHash: Hash): Promise<boolean> => {
     return web3.eth.getTransaction(txHash)
         .then((trx) => !!(trx && trx.blockNumber))
 }
+
+export const averageBlockTime = async (web3Instance: StreamrWeb3, blocksAgo: number = 500) => {
```

Comment:
Just curious, where does `500` come from?

Reply:
It's a magic number, basically. For ethereum main net, there are about 6000 blocks mined per day (https://etherscan.io/chart/blocks). Assuming that, on the main net this would cover the last 2 hours.

Since the main net number is fairly constant, the whole function could actually be replaced with the number 15s (https://etherscan.io/chart/blocktime) :) But this should work in every net